lib/index.js

```diff
@@ -0,0 +1,166 @@
+// #POSTHTML - INCLUDE - INDEX
+// ------------------------------------
+
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+
+const cache = require('read-cache')
+
+/**
+ * @name include
+ * @type {Object}
+ *
+ */
+const include = {
+  /**
+   * Sync Mode
+   *
+   * @memberof include
+   * @method sync
+   *
+   * @param  {String} src File
+   *
+   * @return {Object}     PostHTML Tree
+   */
+  sync: function (node, src, parser) {
+    src = cache.sync(src, 'utf8')
+
+    node.tag = false
+    node.content = parser(src)
+
+    return node
+  },
+  /**
+   * Async Mode
+   *
+   * @memberof include
+   * @method async
+   *
+   * @param  {String} src File
+   *
+   * @return {Promise}    PostHTML Tree
+   */
+  async: function (node, src, parser) {
+    return cache(src, 'utf8')
+      .then((src) => parser(src))
+      .then((content) => {
+        node.tag = false
+        node.content = content
+
+        return node
+      })
+  }
+}
+
+/**
+ * @author Ivan Voischev <voischev.ivan@ya.ru>
+ * @description Include Plugin
+ *
+ * @module posthtml-include
+ * @version 2.0.0
+ *
+ * @requires read-cache
+ * @requires posthtml-parser
+ *
+ * @param  {Object} options Options
+ *
+ * @return {Function}       Include Plugin
+ */
+module.exports = (options) => {
+  options = options || { root: false }
+
+  /**
+   * @method posthtmlInclude
+   *
+   * @param  {Object} tree  PostHTML Tree
+   *
+   * @return {Object} tree  PostHTML Tree (Transformed)
+   */
+  return function posthtmlInclude (tree) {
+    if (!tree.messages) tree.messages = []
+
+    let dir
+
+    tree.options.from
+      ? dir = options.root
+        ? path.join(path.dirname(tree.options.from), options.root)
+        : path.dirname(tree.options.from)
+      : dir = options.root
+        ? path.join(process.cwd(), options.root)
+        : process.cwd()
+
+    const includes = []
+
+    tree.match([ { tag: 'include' }, { tag: 'import' } ], (node) => {
+      let src, err
+
+      if (!node.attrs) {
+        err = {
+          type: 'error',
+          message: 'posthtml-include: Include with missing src found'
+        }
+
+        tree.messages.push(err)
+
+        return node
+      }
+
+      if (node.attrs.src) {
+        src = path.resolve(dir, node.attrs.src)
+      }
+
+      if (!fs.existsSync(src)) {
+        err = {
+          type: 'error',
+          message: `posthtml-include: ${src} could not be loaded`
+        }
+
+        src = false
+
+        tree.messages.push(err)
+
+        return node
+      }
+
+      if (src) {
+        let parser
+
+        if (path.extname(src) !== '.html') {
+          if (!tree.options.parser) {
+            err = {
+              type: 'error',
+              message: `posthtml-include: Parser for ${path.extname(src)} files not specified`
+            }
+
+            return tree.messages.push(err)
+          }
+
+          parser = tree.options.parser
+        }
+
+        if (path.extname(src) === '.html') {
```

Comment:
And, please, move `require('posthtml-parser`)` to the top of document. Because this is create Synchronous request to FS for the first file that needs to be included by this plugin.